extensions/favicon-generator/src/check-favicon.tsx

```diff
@@ -0,0 +1,64 @@
+import { Action, ActionPanel, Form, LaunchProps, showToast, Toast } from "@raycast/api";
+import { checkFavicon } from "./favicon-generator";
+
+export default function Command(
+  props: LaunchProps<{ arguments: Arguments.CheckFavicon; launchContext: { htmlSnippets: string } }>,
+) {
+  const { port } = props.arguments;
+  const { htmlSnippets } = props.launchContext ?? {};
+
+  const handleSubmit = async function handleSubmit(values: { port: string }) {
+    try {
+      const toast = await showToast({ style: Toast.Style.Animated, title: "Checking favicons..." });
+
+      const result = await checkFavicon(values.port);
+
+      toast.style = Toast.Style.Success;
+      toast.title = "Success!";
+      toast.message = "Favicon check completed. Check the logs for results.";
+
+      // Log the detailed results
+      console.log(result);
+    } catch (error) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: "Error checking favicons",
+        message:
+          error instanceof Error && error.message.includes("undici")
+            ? "No web app found on that port"
+            : error instanceof Error
+              ? error.message
+              : "Unknown error occurred",
+      });
+    }
+  };
```

Comment:
You might need some validation before running this. See https://developers.raycast.com/api-reference/user-interface/form#validation.

I can execute the `handleSubmit` even when some fields are empty.